app/forms/DateForm.scala

```diff
@@ -62,6 +62,22 @@ object DateForm extends InputFilters {
     }
   }
 
+  def validateEndDate(formData: DateFormData,
+                      taxYear: Int,
+                      benefitType: BenefitType,
+                      isAgent: Boolean,
+                      startDate: LocalDate)
+                     (implicit messages: Messages): Seq[FormError] = {
+    lazy val emptyDateFieldsValidationValue = emptyDateFieldsValidation(formData, benefitType, datePageName = "endDatePage", isAgent)
+    lazy val invalidDateFormatValidationValue = invalidDateFormatValidation(formData, benefitType, datePageName = "endDatePage", isAgent)
+    lazy val endDateSpecificValidationValue = endDateSpecificValidation(formData.toLocalDate.get, taxYear, benefitType, isAgent, startDate)
```

Comment:
I would rename these `Seq[FormError]` vals to `emptyDateFieldsErrors`, `invalidDateFormatErrors`, and `endDateSpecificErrors` to make it clearer